datahub_sap_hana/ingestion.py

```diff
@@ -178,3 +203,178 @@ def _get_view_lineage_workunits(self) -> Iterable[MetadataWorkUnit]:
                 wu = item.as_workunit()
                 self.report.report_workunit(wu)
                 yield wu
+
+    def get_column_lineage_view_definitions(self, conn) -> Iterable[Tuple[str, str]]:
+        with conn:
+            inspector = inspect(conn)
+            schema: List[Any] = inspector.get_schema_names()  # returns a list
+
+            for schema_name in schema:
+                views = inspector.get_view_names(schema=schema_name)  # returns a list
+
+                for view_name in views:
+                    view_sql = inspector.get_view_definition(view_name, schema_name)
+
+                    if view_sql:
+                        yield (view_name, view_sql)
+
+    def _get_column_lineage_for_view(
+        self, view_name: str, view_sql: str
+    ) -> Tuple[str, List[Node]]:
+        """Extracts the columns and the sql definitions of a downstream view.
+        These are then passed to the lineage function create a lineage graph of the column of an sql query.
+        """
+
+        expression: DerivedTable = parse_one(view_sql)
+        selected_columns: List[str] = expression.named_selects
+
+        view_sql = view_sql.lower()
+        lineages = [
+            lineage(col_name.lower(), view_sql) for col_name in selected_columns
+        ]
+        return view_name, lineages
+
+    def parse_column_name(self, column_name: str):
+        """Removes the table/name and alias from the column name returned by sqlglot, eg: instead of 'hotel.room', it will return 'room'."""
+
+        parsed_name = column_name.split(".")
+        return parsed_name[1]
```

Comment:
there is a potential bug here, the string might not contain a `.`, in that situation you `parsed_name = ['field_name']`, it only contains a single value, so `parsed_name[1]` will produce an error. You can use `parsed_name[-1]` which always works and means "the last element"